### PR TITLE
[ORCA] Fix bug checking index_can_return()

### PR DESCRIPTION
**Pull request by dgkimura**
ORCA uses index_can_return() to check whether an index can return a
given column's data. The result determines whether an index-only-scan
plan is allowed. The function uses attno, however ORCA erroneously
passed the column number instead. This caused ORCA to incorrectly allow
an index-only-scan plan.

See get_relation_info() usage of index_can_return() attno for PLANNER.

### CODE REVIEW

File: src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp [hunk 1]
```diff
@@ -1039,27 +1039,27 @@ CTranslatorRelcacheToDXL::RetrieveIndex(CMemoryPool *mp,
 
 	for (int i = 0; i < form_pg_index->indnatts; i++)
 	{
-		INT attno = form_pg_index->indkey.values[i];
-		GPOS_ASSERT(0 != attno && "Index expressions not supported");
```
Comment (gpopt) [inline src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp:1043]: Can we keep `attno` here? I see infer_arbiter_indexes() (L1051, 1192) also uses attno.

**Review by DevChattopadhyay — APPROVED** [review]
The changes look good to me.

File: src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp (hunk 1 above)
Comment (dgkimura) [reply src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp]: Yeah. Updated accordingly.

**Comment by dgkimura** [comment]
Merged. Thanks for the reviews.

**Pull request merged**